Cleanup `/test/rocmlir-driver/misc_e2e`

zhanglx13 opened the pull request:
This PR cleans up the `test/rocmlir-driver/misc_e2e` dir

- `conv2d_harness_cpu_verifier_kyxc_nhwc_nhwk.mlir` and `conv2d_harness_cpu_verifier_yxck_nhwc_nhwk.mlir` are moved into `test/rocmlir-driver/conv2d_harness/cpu_verifier/`
- The test in `conv2d_harness_cpu_verifier_kyxc_nhwc_nhwk_f16.mlir` is added into `test/e2e/conv2d_regression_fwd.toml`.
- Tests in `conv2d_host_validation.mlir` and `conv2d_host_validation_f16.mlir` (both `RUN` and `FIXME`) are added into `test/e2e/Resnet101Config.toml`, which is enabled as other ROCK E2E tests by `ROCK_E2E_TEST_ENABLED`.
- Tests in `conv2d_host_validation_random.mlir` are added into `test/e2e/conv2d_regression_fwd.toml`.
- Get rid of `ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED`
- Delete `misc_e2e` dir.

Note that some tests in `misc_e2e` have `-rand_type float` as random inputs. This issue will be addressed in https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/869.

zhanglx13 committed:
Convert misc_e2e/conv2d_host_validation[_f16].mlir into
Resnet101Config.toml

Let's see if the FIXME tests can pass

krzysz00 commented on mlir/test/e2e/CMakeLists.txt:
```diff
@@ -40,6 +40,11 @@ if (ROCK_E2E_TEST_ENABLED)
     conv2d_wrw_perf_config
   )
 endif()
+if (ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED)
```
I'd argue this is a good time to get rid of `ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED`

zhanglx13 replied:
Previously, these tests are only enabled with fixed inputs. I'll add them as regular e2e tests and see if they can pass with random inputs.

krzysz00 replied:
Yeah, worth checking. Could be this was an old bug or something.

zhanglx13 committed:
Clean up tests in misc_e2e

- cpu_verifier_*.mlir are moved into
/rocmlir-driver/conv2d_harness/cpu_verifier/
- conv2d_host_validation_random.mlir are put into
conv2d_regression_fwd.toml
- Get rid of ROCMLIR_DRIVER_MISC_E2E_TEST_ENABLED
- remove the misc_e2e dir

Two tests in conv2d_host_validation_random.mlir are using -rand_type
float. The clean up is complete when the float-random-input issue is
resolved in https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/869

krzysz00 reviewed (approved):
Seems fine to me

However, IIRC, there were some Resnext101 configs that were identical to Resnet50 and so got commented out. If those comments were in the files you deleted, can you preserve them in the config file?

zhanglx13 commented:
I did not touch any Resnext101 and Resnet50 tests. Some tests in misc_e2e dir are from Resnet101, which I assume is different from Resnext101.
The configs are commented out in Resnext101Config.toml if they also appear in Resnet50Config.toml.

krzysz00 commented:
Ah. I'm fantastic an noticing that sort of difference.

And so, without previous reservations, LGTM
